test.py

```diff
@@ -202,16 +204,20 @@ def other_vertical_iteration_variable(a: Field[Edge], b: Field[Edge]):
 
 
 @stencil
-def temp_field_demoted(a: Field[Edge], b: Field[Edge], out: Field[Edge]):
+def temp_field_demoted(a: Field[Edge, K], b: Field[Edge, K], out: Field[Edge, K]):
     x: Field[Edge]
+    y: Field[Edge, K]
     with levels_downward:
         x = a + b
```

Comment:
Interesting that dawn is happy with this. `x` doesn't have a vertical dimension. I guess it recognizes that this is actually a vertical region which has to be sequential.

Reply:
It's very debatable if this dusk code should be legal in my opinion. what happens in dawn is that x is demoted to a local variable
```
          for (auto const &loc : getEdges(LibTag{}, m_mesh)) {
            ::dawn::float_type __local_x_3_33 =
                (m_a(deref(LibTag{}, loc), k + 0) +
                 m_b(deref(LibTag{}, loc), k + 0));
            if ((__local_x_3_33 > (int)3)) {
              m_out(deref(LibTag{}, loc), k + 0) = __local_x_3_33;
            } else {
              m_out(deref(LibTag{}, loc), k + 0) = (int)5;
            }
          }
```
hence the codegen works out. An interesting case for sure.

Reply:
That's interesting. A temporary 2d field could be abused like so:
```python
def column_sum(x: Field[Edge, K]):
  sum: Field[Edge]
  with levels_upward[:0]: # bottom level
    sum = x
  with levels_upward[1:]: # remaining levels
    sum = sum + x
```
Would this work?